Fastlz update

murphyzhao opened the pull request:
1.  增加例程   
2.  增加 readme

murphyzhao committed:
[add] add sample

1.  add sample
2.  update SConscript

Signed-off-by: MurphyZhao <d2014zjt@163.com>

murphyzhao committed:
[add] add README.md

Signed-off-by: MurphyZhao <d2014zjt@163.com>

BernardXiong commented on README.md:
```diff
@@ -0,0 +1,55 @@
+`git clone https://github.com/RT-Thread-packages/fastlz.git`
+
+- env工具辅助下载：
+  menuconfig package path：`RT-Thread online package` -> `miscellaneous package` -> `Fastlz`
```
获取方式这部分，不建议直接再给出 `git clone ...git` 的模式，而是着重在menuconfig中怎么使能。

另外，未来有关于menuconfig的描述可以更类似这样：

    RT-Thread online packages --->
        miscellaneous packages --->
            [*] Fastlz

以前面4个空格方式代表一级级选项，同时 `[*]` 代表选择了这样的选项。（注意前面的选项是和menuconfig中是一致的。例如是`RT-Thread online packages`而不是`RT-Thread online package`）

BernardXiong commented on README.md:
```diff
@@ -0,0 +1,55 @@
+### 4.1 解压缩假死
+
+  问题： 在 `msh` 中使用命令进行解压缩的时候， 会遇到解压缩过程中出现 `死机` 的情况  
+  原因： FastLZ 代码中默认配置使用的内存空间比较大， 可能会超出 `msh` 线程的堆栈空间  
```
如果这个栈空间需要到40960，这个本身就不是一个靠谱的例子了，请找出原因出来。

BernardXiong commented on SConscript:
```diff
@@ -1,11 +1,15 @@
-
+Import('RTT_ROOT')
 from building import *
+Import('rtconfig')
```
请多多学习SConscript的写法。

1. Import('RTT_ROOT') 这类，如果在SConscript中并未用到RTT_ROOT变量，那么就不应该导入这个变量；
2. Import('rtconfig') 同上。

BernardXiong commented on SConscript:
```diff
@@ -1,11 +1,15 @@
+if GetDepend('FASTLZ_USING_SAMPLE'):
+    src += ['fastlz_sample.c']
+
+group = DefineGroup('fastlz', src, depend = ['FASTLZ_USING_SAMPLE'], CPPPATH = CPPPATH)
```
请学习下，`python $RTT_ROOT/tools/wizard.py --component TestCom`产生的SConscript脚本，为什么是这样而不是其他。

我们需要保持代码简洁的初心（小而精练，整洁）。

hellozimo replied:
已经修改，看下是否符合要求

BernardXiong commented on fastlz_sample.c:
```diff
@@ -0,0 +1,269 @@
+
+#include <finsh.h>
+
+MSH_CMD_EXPORT_ALIAS(fastlz_test, fastlz_test, fastlz compress and decompress test);
```
这里为什么使用MSH_CMD_EXPORT_ALIAS，而不是MSH_CMD_EXPORT？

hellozimo replied:
已经修改

BernardXiong reviewed (changes requested):
Please change code according to the comments.

armink commented on fastlz_sample.c:
```diff
@@ -0,0 +1,269 @@
+#define free       rt_free
+
+#define BLOCK_HEADER_SIZE              4
+#define COMPRESS_BUFFER_PADDING        400
```
这个 buffer padding 是当时使用 quicklz 时， quicklz 强制要求的。检查下 fastlz 是否需要

hellozimo replied:
需要有5%的余量

The output buffer must be at least 5% larger than the input buffer

murphyzhao committed:
[update]
1. Fix SConscript problems
2. Modify the source of fastlz.c, solve the problem of large static memory
3. update readme file

murphyzhao committed:
[update] 修改 readme 文件中的错误

murphyzhao committed:
[doc/image] 移除 MSH_MEM.jpg
